tokio-executor/src/lib.rs

```diff
@@ -56,6 +56,9 @@
 //! [`DefaultExecutor`]: struct.DefaultExecutor.html
 //! [`Park`]: park/index.html
 //! [`Future::poll`]: https://doc.rust-lang.org/std/future/trait.Future.html#tymethod.poll
+#[cfg(any(feature = "current-thread", feature = "threadpool"))]
+#[macro_use]
+mod tracing;
```

Comment:
Why are you using `macro_use` here? afaik, that is deprecated?

Reply:
`macro_use` is still required for macros defined in a local module.

> Note that you'll still need `#[macro_use]` to use macros you've defined in your own crate; this feature only works for importing macros from external crates.
– [The Edition Guide](https://doc.rust-lang.org/edition-guide/rust-2018/macros/macro-changes.html#macro_rules-style-macros)

Reply:
(the macros are defined in a local module since we wrap `tracing`'s macros in order to allow disabling the `tracing` dependency entirely: we have versions of the macros we use which either expand to the corresponding `tracing` macro when the dependency is enabled, or expand to nothing when the dependency is disabled)

Reply:
Ah, yes...  ok